app/controllers/doorkeeper/authorizations_controller.rb

```diff
@@ -49,10 +50,21 @@ def render_error
       end
     end
 
+    def can_authorize_response?
+      matching_token? && pre_auth.client.application.confidential? && Doorkeeper.config.custom_access_token_attributes.empty?
+    end
+
+    def prepare_custom_attributes
+      return if Doorkeeper.config.custom_access_token_attributes.empty?
+      return if (token = matching_token?).blank?
+      custom_attributes = token.attributes.symbolize_keys.slice(*Doorkeeper.config.custom_access_token_attributes)
+      pre_auth.custom_access_token_attributes = custom_attributes
+    end
```

Comment:
Is there a reason we are not setting this in the `OAuth::PreAuthorization` directly? It seems like logic that should not be in the controller.